### ISSUE TITLE
Sticky group separator shows wrong values

### ISSUE THREAD

**Issue by CripyIce** [Dimibe/grouped_list#14]
I have an issue when displaying a list with contacts names in Hebrew, the value inside the sticky group separator gets kind of stuck and not displaying the correct value. 

Example:
![demo-gif](https://user-images.githubusercontent.com/26387823/77449946-49c3a980-6dfb-11ea-9464-b7c0a8426dcc.gif)

* The first letter of the names in the list is in the right.

**Comment by Dimibe** [comment]
I'll have a look into it and fix it as soon as possible.

**Comment by CripyIce** [comment]
@Dimibe Thanks! It would be great!
Let me know if you need anything else from me.

**Comment by Dimibe** [comment]
@CripyIce I can't reproduces the bug. Can you provide me more information?
* What is the parent widget of the GroupedListView? 
* What language is used in your example?

Thanks.

**Comment by CripyIce** [comment]
@Dimibe here is a full example for testing:
```dart
class MyHomePage extends StatefulWidget {
  MyHomePage({Key key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _contacts = [
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'שיראל',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
    'איתן',
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Grouped List Test"),
      ),
      body: Scrollbar(
        child: GroupedListView<String, String>(
          elements: _contacts,
          itemBuilder: (context, element) {
            return Padding(
              padding: const EdgeInsets.symmetric(vertical: 8.0),
              child: ListTile(
                title: Text(element),
              ),
            );
          },
          order: GroupedListOrder.ASC,
          groupBy: (element) => element.substring(0, 1),
          separator: Divider(
            height: 0.0,
            indent: 16.0,
            endIndent: 16.0,
          ),
          useStickyGroupSeparators: true,
          groupSeparatorBuilder: (value) => Container(
            padding:
                const EdgeInsets.symmetric(vertical: 8.0, horizontal: 16.0),
            width: MediaQuery.of(context).size.width,
            color: Color(0xffF7F7F7),
            child: Text(
              value,
              style: TextStyle(
                fontSize: 18.0,
                color: Colors.red,
                fontWeight: FontWeight.bold,
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

flutter doctor:
```
[✓] Flutter (Channel beta, v1.15.17, on Mac OS X 10.15.4 19E266, locale en-IL)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.6)
[!] VS Code (version 1.43.1)
    ✗ Flutter extension not installed; install from
      https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
[✓] Connected device (3 available)
```

**Comment by CripyIce** [comment]
@Dimibe just updated to 2.0.2 and the issue still appears but now differently.
I will explain:
1. scroll al the way down to `שיראל` - group is correct.
2. scroll up until you see `איתן` - group is not correct - still showing the letter `ש`.
3. scroll up again until almost at the top - group changed and now is correct.

**Comment by Dimibe** [comment]
@CripyIce The bug wasn't fixedl in the Releases 2.0.1 and 2.0.2. 
As you can see the pull request is still open. I need to do some testing. 
I will release a new version in the next few days.

**Comment by talamaska** [comment]
I too can confirm that the sticky header is showing wrong value occasionally.

**Comment by Dimibe** [comment]
@CripyIce @talamaska The Issue is fixed in version 2.1.0 which should now be available.

**Comment by CripyIce** [comment]
@Dimibe thanks for the fix!
But, please see my new issue after upgrading to latest version - #23.
